src/BloomExe/ImageProcessing/ImageServer.cs

```diff
@@ -80,7 +80,11 @@ protected override bool ProcessRequest(IRequestInfo info)
 				r = r.Replace("thumbnail", "");
 				if (File.Exists(r))
 				{
-						info.ReplyWithImage(_cache.GetPathToResizedImage(r));
+					// thumbnail requests have the thumbnail parameter set in the query string
+					var thumb = info.GetQueryString()["thumbnail"] != null;
+					var pathToFile = _cache.GetPathToResizedImage(r, thumb);
+					if (!string.IsNullOrEmpty(pathToFile))
+						info.ReplyWithImage(pathToFile);
 					return true;
```

Comment:
True means the request was handled. I put in code to return false if no file name is returned from GetPathToResizedImage().